### PR TITLE
[flutter_tools] ensure kernel paths match between init from dill and persist

### PR DESCRIPTION
**Pull request by jonahwilliams**
The usage of --flutter-widget-cache only in the initialize from dill argument would cause the dill caching to get out of sync. Filter out experiments and the widget cache option when computing the location for the dill cache file - as these do not affect the dill contents.

### CODE REVIEW

**Comment by jonahwilliams** [comment]
Uh, explained better - what can happen is that the place the compiler trys to look for the initialize from dill file is different than where the resident runner saves that dill sometimes

File: packages/flutter_tools/lib/src/bundle.dart [hunk 1]
```diff
@@ -32,8 +32,10 @@ String getDefaultCachedKernelPath({
   Config? config,
 }) {
   final StringBuffer buffer = StringBuffer();
+   final List<String> cacheFrontEndOptions = extraFrontEndOptions.toList()
+    ..removeWhere((String arg) => arg.startsWith('--enable-experiment=') || arg == '--flutter-widget-cache');
```
Comment (zanderso) [inline packages/flutter_tools/lib/src/bundle.dart:36]: nit: check indentation.

**Review by zanderso — APPROVED** [review]

File: packages/flutter_tools/lib/src/bundle.dart (hunk 1 above)
Comment (jonahwilliams) [reply packages/flutter_tools/lib/src/bundle.dart]: fixed

**Comment by zanderso** [comment]
FYI Skiaperf picked up a small regression on this change:

https://flutter-flutter-perf.skia.org/e/?begin=1613585464&end=1626193899&keys=X74faca70185a9fed7b82f09b0d2fbf7d&requestType=0&xbaroffset=24779

But it looks like maybe we have to accept it since this is a needed bug fix IIUC.

**Pull request merged**